Missing yaml-dev dependency in devcontainer

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Kong version (`$ kong version`)

3.1.0

### Current Behavior

When use devcontainer to prepare develop environment, postCommand `make dev` will fail.

Here is the log:
```bash

Checking stability of dependencies in the absence of
pgmoon 1.13.0-1...

Removing pgmoon 1.13.0-1...
Removal successful.
kong 3.1.0-0 depends on luatz 0.4 (0.4-1 installed)
kong 3.1.0-0 depends on lua_system_constants 0.1.4 (0.1.4-0 installed)
kong 3.1.0-0 depends on lyaml 6.2.8 (not installed)
Installing https://luarocks.org/lyaml-6.2.8-1.src.rock

Error: Failed installing dependency: https://luarocks.org/lyaml-6.2.8-1.src.rock - Could not find library file for YAML
  No file libyaml.a in /usr/local/lib
  No file libyaml.a in /usr/local/lib64
  No file libyaml.so in /usr/local/lib
  No file libyaml.so in /usr/local/lib64
  No file matching libyaml.so.* in /usr/local/lib
  No file matching libyaml.so.* in /usr/local/lib64
  No file libyaml.a in /usr/lib
  No file libyaml.a in /usr/lib64
  No file libyaml.so in /usr/lib
  No file libyaml.so in /usr/lib64
  No file matching libyaml.so.* in /usr/lib
  No file matching libyaml.so.* in /usr/lib64
  No file libyaml.a in /lib
  No file libyaml.a in /lib64
  No file libyaml.so in /lib
  No file libyaml.so in /lib64
  No file matching libyaml.so.* in /lib
  No file matching libyaml.so.* in /lib64
You may have to install YAML in your system and/or pass YAML_DIR or YAML_LIBDIR to the luarocks command.
Example: luarocks install lyaml YAML_DIR=/usr/local
make: *** [Makefile:103: install] Error 1
[30372 ms] postCreateCommand failed with exit code 2. Skipping any further user-provided commands.
Done. Press any key to close the terminal.
```

### Expected Behavior

The devcontainer build success and no errors

### Steps To Reproduce

```markdown
1. Clone source code
2. Open source code with VS Code
3. Open project in decontainer
```


### Anything else?

_No response_

--- a/.devcontainer/Dockerfile
+++ b/.devcontainer/Dockerfile
@@ -6,4 +6,5 @@ RUN apk add --update \
         alpine-sdk \
         build-base \
         bsd-compat-headers \
-        m4
+        m4 \
+        yaml-dev

